### PR TITLE
chore: explicitly reject share commitments that are not == hash len

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

Noticed this while I was writing the block validity rules. This was still failing as expected since the commitment is check, but now we're being more explicit.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: pkg/appconsts/global_consts.go [hunk 1]
```diff
@@ -76,9 +76,17 @@ var (
...
+	hashLength = NewBaseHashFunc().Size()
+
 	// DefaultCodec is the default codec creator used for data erasure.
 	DefaultCodec = rsmt2d.NewLeoRSCodec
 
 	// SupportedShareVersions is a list of supported share versions.
 	SupportedShareVersions = []uint8{ShareVersionZero}
 )
+
+// HashLen returns the length of a hash in bytes.
```
Comment (rootulp) [inline pkg/appconsts/global_consts.go:89]:
```suggestion
// HashLength returns the length of a hash in bytes.
```

File: x/blob/types/payforblob_test.go [hunk 2]
```diff
@@ -147,6 +147,10 @@ func TestValidateBasic(t *testing.T) {
 	emptyShareCommitment := validMsgPayForBlobs(t)
 	emptyShareCommitment.ShareCommitments[0] = []byte{}
 
+	// MsgPayForBlobs that has an empty share commitment
```
Comment (rootulp) [inline x/blob/types/payforblob_test.go:150]:
```suggestion
	// MsgPayForBlobs that has an invalid share commitment size
```

File: pkg/appconsts/global_consts.go (hunk 1 above)
Comment (evan-forbes) [reply pkg/appconsts/global_consts.go]: thaaaaannnkksss

**Review by rootulp — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]
LGTM. I don't know what our stance is towards clearer error messages. I can imagine returning `ErrInvalidShareCommitment` in this scenario would be relatively confusing to users. It would be also helpful now we have multiple share commitments to return which one was invalid by wrapping the error

**Pull request merged**